recipes/ncbi-cxx-toolkit-public/26.0/conanfile.py

```diff
@@ -0,0 +1,1587 @@
+    topics = ("ncbi", "biotechnology", "bioinformatics", "genbank", "gene",
+              "genome", "genetic", "sequence", "alignment", "blast",
+              "biological", "toolkit", "c++")
+    settings = "os", "compiler", "build_type", "arch"
+
+    tk_tmp_tree = ""
+    tk_src_tree = ""
+    all_NCBI_requires = {}
+
+    options = {
+        "shared":     [True, False],
+        "sharedDeps": [True, False],
+        "fPIC":       [True, False],
+        "projects":   "ANY",
+        "targets":    "ANY",
+        "tags":       "ANY"
+    }
+    default_options = {
+        "shared":     False,
+        "sharedDeps": False,
+        "fPIC":       True,
+        "projects":   "",
+        "targets":    "",
+        "tags":       ""
+    }
+    NCBI_to_Conan_requires = {
+#                Linux                  Macos               Windows
+#"BACKWARD":     ("backward-cpp/1.6",    "",                 ""),
+"BACKWARD":     (None,                  None,               None),
+"BerkeleyDB":   ("libdb/5.3.28",        "libdb/5.3.28",     None),
+#"Boost":        ("boost/1.76.0",        "boost/1.76.0",     "boost/1.76.0"),
+"BZ2":          ("bzip2/1.0.8",         "bzip2/1.0.8",      "bzip2/1.0.8"),
+"CASSANDRA":    ("cassandra-cpp-driver/2.15.3", None,       None),
+"GIF":          ("giflib/5.2.1",        "giflib/5.2.1",     "giflib/5.2.1"),
+#"GRPC":         ("grpc/1.39.1",         "grpc/1.39.1",      "grpc/1.39.1"),
+"JPEG":         ("libjpeg/9d",          "libjpeg/9d",       "libjpeg/9d"),
+"LMDB":         ("lmdb/0.9.29",         "lmdb/0.9.29",      "lmdb/0.9.29"),
+"LZO":          ("lzo/2.10",            "lzo/2.10",         "lzo/2.10"),
+"MySQL":        ("libmysqlclient/8.0.25", "libmysqlclient/8.0.25", "libmysqlclient/8.0.25"),
+"NGHTTP2":      ("libnghttp2/1.45.1",   "libnghttp2/1.45.1","libnghttp2/1.45.1"),
+"PCRE":         ("pcre/8.45",           "pcre/8.45",        "pcre/8.45"),
+"PNG":          ("libpng/1.6.37",       "libpng/1.6.37",    "libpng/1.6.37"),
+#"PROTOBUF":     ("protobuf/3.17.1",     "protobuf/3.17.1",  "protobuf/3.17.1"),
+"SQLITE3":      ("sqlite3/3.36.0",      "sqlite3/3.36.0",   "sqlite3/3.36.0"),
+"TIFF":         ("libtiff/4.3.0",       "libtiff/4.3.0",    "libtiff/4.3.0"),
+"XML":          ("libxml2/2.9.12",      "libxml2/2.9.12",   "libxml2/2.9.12"),
+"XSLT":         ("libxslt/1.1.34",      "libxslt/1.1.34",   "libxslt/1.1.34"),
+"UV":           ("libuv/1.42.0",        "libuv/1.42.0",     "libuv/1.42.0"),
+"Z":            ("zlib/1.2.11",         "zlib/1.2.11",      "zlib/1.2.11"),
+    }
+    Conan_package_options = {
+        "libnghttp2": {
+            "with_app": False,
+            "with_hpack": False
+        },
+        "grpc": {
+            "cpp_plugin": True,
+            "csharp_plugin": False,
+            "node_plugin": False,
+            "objective_c_plugin": False,
+            "php_plugin": False,
+            "python_plugin": False,
+            "ruby_plugin": False
+        }
+    }
```

Comment:
If you are forcing some options in your requirements, Conan will compute a different packageID and the binary package won't be available in ConanCenter. Try to go with the default ones and, if something different is needed, we can have a closer look to it.

Reply:
Why request binaries that I do not need? Defaults install a lot of useless stuff, which only takes disk space. I remember I had problems with defaults. I do not remember what exactly though. Need to test again.

Reply:
In CCI we cannot afford to generate all the combinations for all the options (without considering options we already build more than 100 different configurations). This is why those binaries won't be available. I agree default will retrieve more libraries (and targets) than the ones you need, but the linker should do its job and in the end the resulting binary should be ~the same.

Reply:
Nobody expects that from Center. That is why "--build missing" exists

Reply:
Yes, conflicts.

    ERROR: Conflict in cassandra-cpp-driver/2.15.3:
        'cassandra-cpp-driver/2.15.3' requires 'openssl/1.1.1k' while 'libnghttp2/1.45.1' requires 'openssl/1.1.1l'.
        To fix this conflict you need to override the package 'openssl' in your root package.
    ERROR: Conflict in libnghttp2/1.45.1:
        'libnghttp2/1.45.1' requires 'c-ares/1.17.1' while 'grpc/1.39.1' requires 'c-ares/1.17.2'.
        To fix this conflict you need to override the package 'c-ares' in your root package.

All this because libnghttp2 has default option "with_app=True", and that requires a lot of other packages, which create conflicts. What is easier? resolve conflicts, or change libnghttp2 options?

Reply:
We are not going to run with `--build missing`, it is also not affordable. It will turn some builds into full-graph builds that will take too long... and those generated packages will be missing meta information that is associated with the PR where their recipe revision was created. We opted to activate all the options and create bigger packages suitable for all consumers even if they need to download some extra megabytes, which is more sustainable than building more packages with different options. And a successful build is a requirement for a PR, so the recipe should use default options. Maybe a bit of a chicken-and-egg problem.

Conflicts are the other issue in dependency management, when there is a diamond in the graph with a different version, Conan relies on the user to resolve the conflict.

As far as I can see, conan-center-index will keep working the same regarding these two issues, we haven't found a way to alleviate them and, AFAIK, we are successfully building hundred of different libraries and these have never been a blocker... and your PR is green now, so it is neither that blocker.

If we think about in-house forks of conan-center-index, under a more controlled environment (less configurations) I can imagine different approaches to this problem, but for a general-purpose one... there is no silver bullet I think.

Reply:
Sorry, I do not understand. Should I modify the recipe to use default settings, or can I keep it as is - with modified libnghttp2 settings?

Reply:
If you are not using the default options for `libnghttp2`, then Conan should be computing a packageID that is not available in ConanCenter and your PR will fail (because we don't use `--build missing`) here. From the logs in this PR I see that it is raising `ConanInvalidConfiguration` for all the configurations, it is not building any package, and this is why it is successful.

Reply:
> Nobody expects that from Center. That is why "--build missing" exists

Actually, most first timers do, it's important to have an easy onboarding path :+1: